### PR TITLE
Faster mlflow `list_model_versions`

### PR DESCRIPTION
**Pull request by avishniakov**
## Describe changes
I fixed how we fetch the model version from the MlFlow model registry, by basically injecting sorting into the request and validating if the given MlFlow model version is qualifying filters passed in `list_model_versions`, before casting it to inner object, which will pull the model data, which is the top consumer in this story.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [x] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

## Summary by CodeRabbit

- **Refactor**
	- Enhanced the filtering and ordering logic in model version listing to improve performance and accuracy.

### CODE REVIEW

**Comment by avishniakov** [comment]
@coderabbitai review

**Review by strickvl — APPROVED** [review]
LGTM. what's the speed improvement when connected to cloud version?

**Comment by avishniakov** [comment]
> LGTM. what's the speed improvement when connected to cloud version?

With fix: Step model_trainer has finished in 19.175s.
Without fix: Step model_trainer has finished in 26.761s.

Not a breakthrough, but saving ~8 seconds to fetch latest model version from mlflow in our setting.

File: src/zenml/integrations/mlflow/model_registries/mlflow_model_registry.py [hunk 1]
```diff
@@ -619,48 +641,9 @@
...
-                if model_version.created_at is not None
-                else float("inf"),
-                reverse=True,
-            )
-        # Return the model versions.
-        if count:
-            return model_versions[:count]
         return model_versions
 
     def load_model_version(
```
[the remark this answers is not in the document]
Comment (avishniakov) [reply src/zenml/integrations/mlflow/model_registries/mlflow_model_registry.py]: thx

**Comment by strickvl** [comment]
Nice. Yeah I feel like we underestimate how much slowness of zenml affects our users.

**Pull request merged**